api/src/main/kotlin/edu/wgu/osmt/elasticsearch/SearchController.kt

```diff
@@ -142,6 +151,64 @@ class SearchController @Autowired constructor(
             .body(searchHits.map { it.content }.toList())
     }
 
+    @Transactional(readOnly = true)
+    @PostMapping(RoutePaths.SCROLL_SKILLS, produces = ["text/csv"])
+    @ResponseBody
+    fun scrollSkills(
+        uriComponentsBuilder: UriComponentsBuilder,
+        @RequestParam(
+            required = false, defaultValue = PublishStatus.DEFAULT_API_PUBLISH_STATUS_SET
+        ) status: Array<String>,
+        @RequestParam(required = false) sort: String?,
+        @RequestBody apiSearch: ApiSearch,
+        @AuthenticationPrincipal user: Jwt?
+    ): ResponseEntity<*> {
+        if (!appConfig.allowPublicSearching && user === null) {
+            throw GeneralApiException("Unauthorized", HttpStatus.UNAUTHORIZED)
+        }
+
+        val publishStatuses = status.mapNotNull {
+            val status = PublishStatus.forApiValue(it)
+            if (user == null && (status == PublishStatus.Deleted || status == PublishStatus.Draft)) null else status
+        }.toSet()
+        val pageable = PageRequest.of(0, 1)
+        val responseHeaders = HttpHeaders()
+        responseHeaders.add("Content-Type", "text/csv")
+
+
+        val searchHits: Stream<SearchHit<RichSkillDoc>> = richSkillEsRepo
+            .streamByApiSearch(apiSearch, publishStatuses, pageable, null)
+
+        val jFactory = JsonFactory()
```

Comment:
Do we still need these 3 lines about jFactory? I think we don't but I might be missing something.

Reply:
No we don't, I will remove them. good catch @JohnKallies